src/uharfbuzz/_harfbuzz.pyx

```diff
@@ -409,6 +409,11 @@ cdef class Font:
         cdef Font inst = cls(face)
         return inst
 
+    def create_sub_font(self, parent: Font):
+        cdef Font inst = self.__class__()
```

Comment:
I belive to instantiate a uharfbuzz.Font also you need to pass in a uharfbuzz.Face as well, see `__cinit__` above which takes a positional `face` parameter. Maybe just setting that to `parent.face` should work for your use case.

Also this new `create_sub_font` method would better be decorated with `@classmethod` and take a `cls` as first parameter rathen than `self` (for instance bound methods), since the way you wrote it, it behaves like an alternative constructor.

Actually, you could turn this into an instance method if you like, and then simply use it as `font.create_sub_font()` instead of `Font.create_sub_font(font)`. E.g.:

```cython
    def create_sub_font(self):
        cdef Font inst = self.__class__(self.face)
        inst._hb_font = hb_font_create_sub_font(self._hb_font)
        return inst
```

Reply:
hm my snippet above may leak because I haven't called hb_font_destroy on the hb_font created inside `__cinit__` when I did `self.__class__(self.face)`...

Better to avoid calling `__cinit__` altogether and use `__new__` like so (untested):

```cython
    def create_sub_font(self):
        cdef Font inst = self.__class__.__new__(self.__class__)
        inst._hb_font = hb_font_create_sub_font(self._hb_font)
        inst._face = self._face
        return inst
```

Reply:
Instance method was my intention indeed. I'm undecided as to which one makes most sense.

Reply:
+1 for instance method, even though it's not identical to the HB C-api